### ISSUE TITLE
Error executing `etlas test` with hspec

### ISSUE THREAD

**Issue by jneira** [typelead/eta#655]
* I've setup a test suite for wai-servlet-handler-jetty in [this branch](https://github.com/jneira/wai-servlet-handler-jetty/tree/test-suite)
* I am able to run the test suite but it fails with this exception:
```
Exception in thread "main" eta.runtime.exception.EtaException: withArgv: Cannot
modify arguments.
        at base.ghc.TopHandler.runMainIO2(TopHandler.hs)
        at base.ghc.TopHandler$runMainIO2.apply1V(TopHandler.hs)
        at eta.runtime.exception.Exception.catch_(Exception.java:166)
        at main.Main.main2(Spec.hs)
        at main.Main.DZCmain(Spec.hs:6)
        at main.Main$DZCmain.applyV(Spec.hs:6)
        at eta.runtime.stg.Closures$EvalLazyIO.enter(Closures.java:104)
        at eta.runtime.stg.Capability.schedule(Capability.java:174)
        at eta.runtime.stg.Capability.scheduleClosure(Capability.java:119)
        at eta.runtime.Runtime.evalLazyIO(Runtime.java:170)
        at eta.runtime.Runtime.main(Runtime.java:163)
        at eta.main.main(Unknown Source)
Caused by: eta.runtime.exception.EtaException: withArgv: Cannot modify arguments
.
        at base.ghc.Err.error(Err.hs:35)
        at base.system.Environment$withArgv.thunkEnter(Environment.hs:167)
        at eta.runtime.thunk.CAF.evaluate(CAF.java:35)
        at hspec_core.test.hspec.core.Runner.$wa(Runner.hs:194)
        at main.Main.main1(Spec.hs:6)
        at main.Main$main1.applyV(Spec.hs)
        at eta.runtime.exception.Exception.catch_(Exception.java:129)
        ... 9 more
Test suite spec: FAIL
```

**Comment by rahulmutt** [comment]
@jneira This is a problem @indegro reported a while back. The temporary fix was to eliminate usage of `withArgs` altogether in `hspec-core` and the permanent solution is to actually implement `withArgv` propertly, tracked in `https://github.com/typelead/eta/issues/611`. What I'll do is fix the version of `hspec-core` to be used in `etlas-index` and make sure that version has the patch that temporarily removes `withArgv`.

**Comment by rahulmutt** [comment]
Can you try installing `hspec-core-2.4.6` and see if that works? I've updated `etlas-index` so after running `etlas update`, `etlas` should prefer to use that version.

**Comment by jneira** [comment]
It works! thanks
Just pont out that i had the 2.4.7 already installed and i had to unregister it with eta-pkg

**Comment by Jyothsnasrinivas** [comment]
@jneira I've implemented #611 so with the latest Eta, you should be able to use any version of `hspec`. We will keep the constraint in `etlas-index` until we do a binary release with #611 implemented.
